unixfs/src/walk.rs

```diff
@@ -546,16 +546,16 @@ impl InnerEntry {
     }
 
     fn as_bucket_root(&mut self, cid: Cid, name: &str, depth: usize, metadata: Metadata) {
-        use InnerKind::*;
         match self.kind {
-            RootDirectory(_)
-            | BucketAtRoot(_)
-            | Bucket(_)
-            | RootBucket(_)
-            | Directory(_)
-            | File(_, None, _)
-            | Symlink(_) => {
-                self.kind = RootBucket(cid);
+            InnerKind::Bucket
+            | InnerKind::BucketAtRoot
+            | InnerKind::Directory
+            | InnerKind::File(None, ..)
+            | InnerKind::RootBucket
+            | InnerKind::RootDirectory
+            | InnerKind::Symlink => {
+                self.cid = cid;
+                self.kind = InnerKind::RootBucket;
```

Comment:
While trying to understand the code, some time were spend to figure out where these variants were coming from even with the top level import declaration. This declarative direction has a clearer approach and says right away its intention.

It might be silly but for a file that has more than 1000 lines where most of the time you are thinking: What is going on? Every help counts.

Nevertheless, it is just a nit that can be reverted if it is indeed desired.

Reply:
> While trying to understand the code, some time were spend to figure out where these variants were coming from even with the top level import declaration.

Yeah, I often shuffle code around as well while learning and reading it. I just try to make sure I `git checkout -p` those changes before they end up in my commits; splitting stuff like this out afterwards with `git rebase -i`and `edit` is not impossible but ... It's just easier to do it up front :)

> This declarative direction has a clearer approach and says right away its intention.

I kind of understand this point, and I understand that it's not too often I've seen the enum variants being used however I feel that we are now discussing wheter a specifically added language feature should or should not be used for which it was added for. I don't think this is the forum for such discussions.